### PR TITLE
ceph: allow setting pool size 1 on octopus

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

Left from https://github.com/rook/rook/pull/4895
Also more cleanup on the ceph.conf since we config is in the mon store.

Signed-off-by: Sébastien Han <seb@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[test ceph]

### CODE REVIEW

File: pkg/operator/ceph/config/defaults.go [hunk 1]
```diff
@@ -67,6 +67,13 @@ func DefaultCentralizedConfigs(cephVersion version.CephVersion) []Option {
 		}...)
 	}
 
+	// For Octopus
+	if cephVersion.IsAtLeastOctopus() {
+		overrides = append(overrides, []Option{
+			configOverride("global", "mon allow pool size one", "true"),
```
Comment (travisn) [inline pkg/operator/ceph/config/defaults.go]: Can we just add this overrides without checking the version? Nautilus would ignore settings it doesn't understand, right?

Comment (leseb) [reply pkg/operator/ceph/config/defaults.go]: Nop, the store will report an error for an unknown option.

**Comment by liewegas** [comment]
This sets the new option that is present in ceph *master* but not in the octopus branch.  We either need to backport this option to octopus (@neha-ojha @ideepika?) or we need to make rook only set this on post-octopus versions of ceph.

**Comment by neha-ojha** [comment]
> This sets the new option that is present in ceph _master_ but not in the octopus branch. We either need to backport this option to octopus (@neha-ojha @ideepika?) or we need to make rook only set this on post-octopus versions of ceph.

let's make rook set it only for post-octopus versions

**Comment by leseb** [comment]
> > This sets the new option that is present in ceph _master_ but not in the octopus branch. We either need to backport this option to octopus (@neha-ojha @ideepika?) or we need to make rook only set this on post-octopus versions of ceph.
> 
> let's make rook set it only for post-octopus versions

Oh I thought this was on Octopus. Will fix.

**Pull request merged**